File: LongLib/src/main/java/com/hloong/lib/longlog/base/LongLogConfig.java
package com.hloong.lib.longlog.base;

public abstract class LongLogConfig {
    public static int MAX_LEN = 512;
    public static LongThreadFormatter THREAD_FORMATTER = new LongThreadFormatter();
    public static LongStackTraceFormatter STACK_TRACE_FORMATTER = new LongStackTraceFormatter();

    public String getGlobalTag(){
        return "LongLog";
    }
    public boolean enable(){
        return true;
    }

    public JsonParser injectJsonParser(){
        return null;
    }

    public boolean includeThread(){
        return false;
    }

    /**
     * get stack depth
     * @return default 5
     */
    public int stackTraceDepth() {
        return 5;
    }

    public LongLogPrinter[] printers(){
        return null;
    }

    public interface JsonParser{
        String toJson(Object object);
    }
}

File: LongLib/src/main/java/com/hloong/lib/longlog/base/LongLogConfig.kt
package com.hloong.lib.longlog.base

abstract class LongLogConfig {
    open fun enable(): Boolean {
        return true
    }
    open fun getGlobalTag(): String? {
        return "LongLog"
    }

    open fun injectJsonParser(): JsonParser? {
        return null
    }

    open fun includeThread(): Boolean {
        return false
    }

    /**
     * get stack depth
     * @return default 5
     */
    open fun stackTraceDepth(): Int {
        return 5
    }

    fun printers(): Array<LongLogPrinter>? {
        return null
    }

    interface JsonParser {
        fun toJson(obj: Any?): String?
    }

    companion object {
        @JvmField
        var MAX_LEN = 512
        @JvmField
        var THREAD_FORMATTER = LongThreadFormatter()
        @JvmField
        var STACK_TRACE_FORMATTER = LongStackTraceFormatter()
    }
}